packages/test/src/test-integration-workflows.ts

```diff
@@ -204,3 +209,52 @@ test('Worker allows heartbeating activities after shutdown has been requested',
   await worker.run();
   t.is(cancelReason, 'CANCELLED');
 });
+
+export async function conditionTimeout0Simple(): Promise<boolean | undefined> {
+  let validationTimerFired = false;
+  // Don't lower this value, otherwise, in CI, there is risk that the timer fire
+  // while the WFT is being processed, which would result in an UnhandledCommand.
+  workflow
+    .sleep(10_000)
+    .then(() => (validationTimerFired = true))
+    .catch((e) => console.log(e));
```

Comment:
Why do we need this timer? Isn't it enough to check that condition returned false?